src/products/services/products.service.ts

```diff
@@ -14,11 +16,13 @@ export class ProductsService {
     private readonly productRepository: ProductsRepository,
   ) {}
 
-  async getAll(): Promise<ProductEntity[]> {
+  async getAll(
+    query: PaginationQueryDTO,
+  ): Promise<Pagination<ProductEntity[]>> {
     try {
       this.logger.log(`Attempting to find all products`);
-      const products = await this.productRepository.findAll();
-      this.logger.log(`Founded ${products.length} products`);
+      const products = await this.productRepository.findAll(query);
+      this.logger.log(`Founded ${products.results.length} products`);
```

Comment:
But, Is there any case that products return null? I mean, it will return empty if there is not records; or throw an error that will scale without reaching the next line